controllers/routemonitor/routemonitor_controller_supplement.go

```diff
@@ -168,16 +167,203 @@ func (r *RouteMonitorReconciler) CreateBlackBoxExporterService(ctx context.Conte
+			// return unexpectedly
+			return nil, err
+		}
+		// populate the resource with the template
+		resource := populationFunc()
+		// and create it
+		err = r.Create(ctx, resource)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return nil, nil
+}
+
+/*
+
+This next block is for deletion of the RouteMonitor
+
+*/
+
+func (r *RouteMonitorReconciler) ShouldDeleteBlackBoxExporterResources(ctx context.Context, routeMonitor *v1alpha1.RouteMonitor) (bool, error) {
+	// if a delete has not been requested then there is at least one resource using the BlackBoxExporter
+	if !r.WasDeleteRequested(routeMonitor) {
+		return false, nil
+	}
+
+	routeMonitors := &v1alpha1.RouteMonitorList{}
+	if err := r.List(ctx, routeMonitors); err != nil {
+		return false, err
+	}
+
+	amountOfRouteMonitors := len(routeMonitors.Items)
+	if amountOfRouteMonitors == 0 {
+		err := errors.New("Internal Fault: Cannot be in reconcile loop and have not RouteMonitors on cluster")
+		// the response is set to true as this technically a case where we should delete, but as it's not logical that this will happen, returning error
+		return true, err
+	}
+
+	// If more than one resource exists, and the deletion was requsted there are stil at least one resource using the BlackBoxExporter
+	if amountOfRouteMonitors > 1 {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+func (r *RouteMonitorReconciler) DeleteBlackBoxExporterResources(ctx context.Context) error {
+	// Deleting Service first because:
+	//
+	// a Service should not point to an empty target (Deployment)
+	if err := r.DeleteBlackBoxExporterService(ctx); err != nil {
 		return err
 	}
-	if err := r.CreateBlackBoxExporterService(ctx); err != nil {
+	if err := r.DeleteBlackBoxExporterDeployment(ctx); err != nil {
 		return err
 	}
 	return nil
 }
 
+func (r *RouteMonitorReconciler) DeleteBlackBoxExporterDeployment(ctx context.Context) error {
+	resource := &appsv1.Deployment{}
+
+	// Does the resource already exist?
+	err := r.Get(ctx, blackBoxNamespacedName, resource)
+	if err != nil {
+		// If this is an unknown error
+		if !k8serrors.IsNotFound(err) {
+			// return unexpectedly
+			return err
+		}
+		// Resource doesn't exist, nothing to do
+		return nil
+	}
+	err = r.Delete(ctx, resource)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *RouteMonitorReconciler) DeleteBlackBoxExporterService(ctx context.Context) error {
+	resource := &corev1.Service{}
+
+	// Does the resource already exist?
+	err := r.Get(ctx, blackBoxNamespacedName, resource)
+	if err != nil {
+		// If this is an unknown error
+		if !k8serrors.IsNotFound(err) {
+			// return unexpectedly
+			return err
+		}
+		// Resource doesn't exist, nothing to do
+		return nil
+	}
+	err = r.Delete(ctx, resource)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// DeleteServiceMonitorResourceCommand is purely for testing purposes
+var DeleteServiceMonitorResourceCommand func(context.Context, *v1alpha1.RouteMonitor) error
```

Comment:
If it's only for testing, IMHO it should not be in the production code

Reply:
then how do you test this?